src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -320,19 +320,19 @@ void IDeserializationCallback.OnDeserialization(object sender)
         // Factory methods.
         //
 
-        public static WindowsIdentity GetCurrent()
+        public static WindowsIdentity? GetCurrent()
         {
-            return GetCurrentInternal(TokenAccessLevels.MaximumAllowed, false);
+            return GetCurrentInternal(TokenAccessLevels.MaximumAllowed, false)!;
         }
 
 
-        public static WindowsIdentity GetCurrent(bool ifImpersonating)
+        public static WindowsIdentity? GetCurrent(bool ifImpersonating)
         {
             return GetCurrentInternal(TokenAccessLevels.MaximumAllowed, ifImpersonating);
         }
 
 
-        public static WindowsIdentity GetCurrent(TokenAccessLevels desiredAccess)
+        public static WindowsIdentity? GetCurrent(TokenAccessLevels desiredAccess)
```

Comment:
These three methods signatures are the only ones that have impact in the public API of this library. It is not clear to me whether I should annotate as non-nullable, however it is evident from WindowsPrincipal.cs#L179 that the parameterless method is expected to be non-nullable. I'm not familiar with this component, so not sure what the expected invariant should be here.

Reply:
Notice that `GetCurrentInternal(TokenAccessLevels desiredAccess, bool threadOnly)` can return null only when `threadOnly == true`. This should give you decoder ring which variants of `GetCurrent` can may return null and when.

Reply:
Right, so makes sense to `GetCurrent(bool)` as the only method with nullable return?

Reply:
Correct.